api/notd/block_processor.py

```diff
@@ -59,12 +77,77 @@ async def get_transfers_in_block(self, blockNumber: int) -> List[RetrievedTokenT
+            erc1155TokenTransfers += await asyncio.gather(*[self._process_erc1155_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc1155EventsChunk])
+
+        erc1155BatchTokenTransfers = []
+        erc1155Batchevents = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferBatchEventSignatureHash])
+        logging.info(f'Found {len(erc1155Batchevents)} erc1155Batchevents in block #{blockNumber}')
+        for erc1155BatchEventsChunk in list_util.generate_chunks(erc1155Batchevents, 5):
+            erc1155BatchTokenTransfers += await asyncio.gather(*[self._process_erc1155_batch_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc1155BatchEventsChunk])
+
+        tokenTransferList = [tokenTransfer for tokenTransfer in erc1155BatchTokenTransfers for tokenTransfer in tokenTransfer] + [tokenTransfer for tokenTransfer in erc1155TokenTransfers if tokenTransfer] + [tokenTransfer for tokenTransfer in erc721TokenTransfers if tokenTransfer]
+        return  tokenTransferList
+
+    async def _process_erc1155_event(self, event: LogReceipt, blockNumber: int, blockHash: str, blockDate: datetime.datetime) -> Optional[RetrievedTokenTransfer]:
+        transactionHash = event['transactionHash'].hex()
+        registryAddress = event['address']
+        logging.debug(f'------------- {transactionHash} ------------')
+        if len(event['topics']) < 4:
+            logging.debug('Ignoring event with less than 4 topics')
+            return None
+        operatorAddress = normalize_address(event['topics'][1].hex())
+        fromAddress = normalize_address(event['topics'][2].hex())
+        toAddress = normalize_address(event['topics'][3].hex())
+        data = event['data']
+        data = textwrap.wrap(data[2:],64)
+        data = [literal_eval(f'0x{elem}') for elem in data]
+        tokenId = data[0]
+        amount = data[1]
+        ethTransaction = await self._get_transaction(transactionHash=transactionHash)
+        gasLimit = ethTransaction['gas']
+        gasPrice = ethTransaction['gasPrice']
+        value = ethTransaction['value']
+        ethTransactionReceipt = await self._get_transaction_receipt(transactionHash=transactionHash)
+        gasUsed = ethTransactionReceipt['gasUsed']
+        transaction = ERC1155TokenTransfer(transactionHash=transactionHash, operatorAddress=operatorAddress, registryAddress=registryAddress, fromAddress=fromAddress, toAddress=toAddress, tokenId=tokenId, amount=amount ,value=value, gasLimit=gasLimit, gasPrice=gasPrice, gasUsed=gasUsed, blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate)
+        return transaction
+
+    async def _process_erc1155_batch_event(self, event: LogReceipt, blockNumber: int, blockHash: str, blockDate: datetime.datetime) -> Optional[RetrievedTokenTransfer]:
```

Comment:
return type is wrong here, make it a list and replace return None with return []